pool/boltupgrades_test.go

```diff
@@ -39,8 +39,12 @@ func TestBoltDBUpgrades(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	t.Cleanup(func() {
```

Comment:
I'd suggest just replacing all this and the `os.MkdirTemp` with the following which handles creation, cleanup, and errors:

```suggestion
	d:= t.TempDir()
```